gax-java/gax/src/test/java/com/google/api/gax/rpc/EndpointContextTest.java

```diff
@@ -340,6 +340,21 @@ public void endpointContextBuild_gdchFlow_noUniverseDomain_customEndpoint() thro
         .isEqualTo(Credentials.GOOGLE_DEFAULT_UNIVERSE);
   }
 
+  // This Universe Domain should match the `GOOGLE_CLOUD_UNIVERSE_DOMAIN` Env Var
+  // For this test running locally or in CI, check that the Env Var is set properly.
+  // This test should only run when the maven profile `EnvVarTest` is enabled.
+  @Test
+  public void endpointContextBuild_universeDomainEnvVarSet() throws IOException {
```

Comment:
Can you please add one more test case that if universe domain is set on both ClientSettings and Env Var, then ClientSettings should take precedence.

Reply:
Yep, added a test case that will be run with `-PenvVarTest`